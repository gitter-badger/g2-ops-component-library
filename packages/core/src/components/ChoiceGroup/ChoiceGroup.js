import * as React from 'react'
import { ChoiceGroup as FabricChoiceGroup } from 'office-ui-fabric-react/lib/ChoiceGroup'

export const ChoiceGroup = ({ 
    selected, 
    defaultSelected, 
    options, 
    onChange, 
    label, 
    styles,
    ...otherProps 
}) => (
  <FabricChoiceGroup
    defaultSelectedKey={selected || defaultSelected}
    options={options}
    onChange={onChange}
    label={label}
    styles={styles}
    {...otherProps}
  /> 
)

