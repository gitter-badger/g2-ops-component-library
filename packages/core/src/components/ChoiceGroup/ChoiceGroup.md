ChoiceGroup Example - 

```js
<div>
   <ChoiceGroup
       defaultSelectedKey="B"
       options={[
          {
              key: 'A',
              text: 'Option A',
          },
          {
              key: 'B',
              text: 'Option B'
          },
          {
              key: 'C',
              text: 'Option C',
              disabled: true
          },
          {
              key: 'D',
              text: 'Option D',
              disabled: true
          }
        ]}
        onChange={(event, option) => console.log(option)}
        label="Pick one"
        required={true}
    />
</div>
```