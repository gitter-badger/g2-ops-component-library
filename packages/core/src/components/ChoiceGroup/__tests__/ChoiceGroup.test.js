import React from 'react'
import renderer from 'react-test-renderer'

import { ChoiceGroup } from '../ChoiceGroup'

describe('<ChoiceGroup />', () => {
  test('should render properly', () => {
    const tree = renderer.create(
      <ChoiceGroup
        defaultSelectedKey="B"
        options={[
          {
              key: 'A',
              text: 'Option A',
          },
          {
              key: 'B',
              text: 'Option B'
          },
          {
              key: 'C',
              text: 'Option C',
              disabled: true
          },
          {
              key: 'D',
              text: 'Option D',
              disabled: true
          }
        ]}
        onChange={(event, option) => console.log(option)}
        label="Pick one"
        required
      />
    )
    expect(tree).toMatchSnapshot()
  })
})