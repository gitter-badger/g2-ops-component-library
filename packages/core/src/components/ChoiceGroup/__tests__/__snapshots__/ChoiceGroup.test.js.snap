// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChoiceGroup /> should render properly 1`] = `
<div
  role="application"
>
  <div
    aria-labelledby="ChoiceGroup0-label "
    className="ms-ChoiceFieldGroup root"
    role="radiogroup"
  >
    <label
      className="ms-Label root-33"
      id="ChoiceGroup0-label"
      required={true}
    >
      Pick one
    </label>
    <div
      className="ms-ChoiceFieldGroup-flexContainer"
    >
      <div
        className="ms-ChoiceField choiceField"
      >
        <div
          className="ms-ChoiceField-wrapper choiceFieldWrapper"
        >
          <input
            aria-labelledby="ChoiceGroupLabel1-A"
            checked={false}
            className="ms-ChoiceField-input input"
            id="ChoiceGroup0-A"
            name="ChoiceGroup0"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="radio"
          />
          <label
            className="ms-ChoiceField-field field"
            htmlFor="ChoiceGroup0-A"
          >
            <span
              className="ms-Label"
              id="ChoiceGroupLabel1-A"
            >
              Option A
            </span>
          </label>
        </div>
      </div>
      <div
        className="ms-ChoiceField choiceField"
      >
        <div
          className="ms-ChoiceField-wrapper choiceFieldWrapper"
        >
          <input
            aria-labelledby="ChoiceGroupLabel1-B"
            checked={true}
            className="ms-ChoiceField-input input"
            id="ChoiceGroup0-B"
            name="ChoiceGroup0"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="radio"
          />
          <label
            className="ms-ChoiceField-field field is-checked fieldIsChecked"
            htmlFor="ChoiceGroup0-B"
          >
            <span
              className="ms-Label"
              id="ChoiceGroupLabel1-B"
            >
              Option B
            </span>
          </label>
        </div>
      </div>
      <div
        className="ms-ChoiceField choiceField"
      >
        <div
          className="ms-ChoiceField-wrapper choiceFieldWrapper"
        >
          <input
            aria-labelledby="ChoiceGroupLabel1-C"
            checked={false}
            className="ms-ChoiceField-input input"
            disabled={true}
            id="ChoiceGroup0-C"
            name="ChoiceGroup0"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="radio"
          />
          <label
            className="ms-ChoiceField-field field is-disabled fieldIsDisabled"
            htmlFor="ChoiceGroup0-C"
          >
            <span
              className="ms-Label"
              id="ChoiceGroupLabel1-C"
            >
              Option C
            </span>
          </label>
        </div>
      </div>
      <div
        className="ms-ChoiceField choiceField"
      >
        <div
          className="ms-ChoiceField-wrapper choiceFieldWrapper"
        >
          <input
            aria-labelledby="ChoiceGroupLabel1-D"
            checked={false}
            className="ms-ChoiceField-input input"
            disabled={true}
            id="ChoiceGroup0-D"
            name="ChoiceGroup0"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="radio"
          />
          <label
            className="ms-ChoiceField-field field is-disabled fieldIsDisabled"
            htmlFor="ChoiceGroup0-D"
          >
            <span
              className="ms-Label"
              id="ChoiceGroupLabel1-D"
            >
              Option D
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
